models/model.js

```diff
@@ -53,7 +58,9 @@ exports.getArticles = async (topic) => {
   try {
     if (topic.topic) {
       const query = await connection.query(`
-      SELECT * FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id WHERE topic = '${topic.topic}'
+      SELECT articles.*, COUNT(comments) AS comment_count 
+      FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id WHERE articles.topic = '${topic.topic}'
+      GROUP BY articles.article_id ORDER BY articles.created_at DESC;
```

Comment:
Also, doubling up on SQL queries is not the best idea - it leads to really confusing code blocks! Have a look at how to build up a query conditionally based on the input in these notes: https://notes.northcoders.com/courses/js-back-end/complex-queries#optional-where-clauses